src/main/java/co/elastic/support/diagnostics/chain/DiagnosticContext.java

```diff
@@ -35,7 +38,12 @@
    public List<String> dockerContainers = new ArrayList<String>();
    public Map<String, RestEntry> elasticRestCalls;
 
+   public ResourceCache resourceCache;
 
-
-
+   public DiagnosticContext(DiagConfig diagConfig, DiagnosticInputs diagnosticInputs, ResourceCache resourceCache, Boolean includeLogs) {
```

Comment:
Since we don't intend to allow `null`, we probably shouldn't imply it:

```suggestion
   public DiagnosticContext(DiagConfig diagConfig, DiagnosticInputs diagnosticInputs, ResourceCache resourceCache, boolean includeLogs) {
```